crates/mpc-protocols/src/mpc_service.rs

```diff
@@ -0,0 +1,117 @@
+use crate::mpc_events::{CreatedProofMPCEvent, MPCEvent};
+use log::info;
+use pera_types::base_types::ObjectID;
+use pera_types::event::Event;
+use std::cell::RefCell;
+use std::collections::{HashMap, VecDeque};
+use tokio::sync::{mpsc, Mutex, RwLock};
+
+// TODO (#253): Make this configurable from a config file
+const MAX_ACTIVE_MPC_INSTANCES: usize = 100;
+
+pub enum MPCInput {
+    InitEvent(CreatedProofMPCEvent),
+    Message,
+}
+
+struct MPCInstance {
+    status: MPCSessionStatus,
+    /// The channel to send message to this instance's message handler thread when this instance is active
+    input_receiver: Option<mpsc::Sender<MPCInput>>,
+    pending_messages: Vec<MPCInput>,
+}
+
+/// Possible status of an MPC session
+/// - Active: The session is currently running, new messages will be forwarded to the session
+/// - Pending: The session is waiting for a slot to become active, when the init received there were already `MAX_ACTIVE_MPC_INSTANCES` active sessions
+/// - Finished: The session has finished, no more messages will be forwarded. The session will be removed from the service after a timeout.
+/// We want to keep it for some time so leftover messages related to it won't be treated as malicious.
+#[derive(Clone, Copy)]
+enum MPCSessionStatus {
+    Active,
+    Pending,
+    Finished,
+}
+
+/// The `MPCService` is responsible for managing MPC instances:
+/// - It keeps track of all MPC instances
+/// - Runs the MPC session for each active instance
+/// - Ensures that the number of active sessions does not go over `MAX_ACTIVE_MPC_INSTANCES` at the same time
+pub struct MPCService {
+    mpc_instances: HashMap<ObjectID, MPCInstance>,
+    /// Being used to keep track of the order of the received pending instances, so we'll know which one to launch first.
```

Comment:
Again: over-documentation.

```suggestion
    /// Used to keep track of the order in which pending instances are received so they are activated in order of arrival.
```
I don't even like this being here, but OK.